torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -209,6 +209,11 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
             global_tp_mesh_dim = _mesh_resources.get_parent_mesh_dim(tp_mesh)
             assert global_dp_mesh_dim is not None  # mypy
             assert global_tp_mesh_dim is not None  # mypy
+            # for PP, DP, TP case, dp mesh dim would be 1, tp mesh dim would be 2
+            # DP/TP would only live in the inner most 2-3 dims
+            if dp_global_mesh.ndim > 2:
```

Comment:
nit: it was a bit confusing for me what `dp_global_mesh` means.  I now understand (?) it refers to 'global mesh' from dp_mesh perspective.  But after that validation code above we should probably stop using those names and always use the less confusing name `global_mesh` instead.  Would that be possible or am i missing somethign?

Reply:
Yes that make sense to me! let me adjust it to make naming more clear